crypto/s2n_certificate.c

```diff
@@ -238,8 +260,57 @@ int s2n_send_cert_chain(struct s2n_stuffer *out, struct s2n_cert_chain *chain)
     return 0;
 }
 
-int s2n_send_empty_cert_chain(struct s2n_stuffer *out) {
+int s2n_send_empty_cert_chain(struct s2n_stuffer *out)
+{
     notnull_check(out);
     GUARD(s2n_stuffer_write_uint24(out, 0));
     return 0;
 }
+
+static int s2n_does_cert_san_match_hostname(X509 *cert, const char *hostname)
```

Comment:
This looks like this parsing is done upon receipt of the ClientHello. Can we instead parse the cert SANs once (when s2n_config_add_cert_chain_and_key_to_store) is called to save having to do it every time and in the request path?

Reply:
Right, we can totally just populate a `GENERAL_NAMES *` field in the s2n_certificate struct. I'll make taht update

Reply:
Added it in 
aeb0011

Reply:
and actually fixed in 
867bbe1